modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/MediaSessionManager.kt

```diff
@@ -171,6 +176,15 @@ class MediaSessionManager(
             .doOnNext { updateUpNext(it) }
             .subscribeBy(onError = { Timber.e(it) })
             .addTo(disposables)
+        settings.useRssArtwork.flow.asObservable(coroutineContext)
```

Comment:
We don't use RxJava for new code. This should be implemented using `Flow` from coroutines. And since my previous suggestion mentions combining this stream with `upNextQueue` stream you'll need to convert `upNextQueue` to `Flow` as well.

Reply:
Any thoughts on how to convert `upNextQueue` to a `Flow`? I implemented a new method in `UpNextQueue`
```kotlin
fun getChangesFlowWithLiveCurrentEpisode(episodeManager: EpisodeManager, podcastManager: PodcastManager): Flow<State> {
    return changesObservable.asFlow().debounce(100).flatMapLatest { state ->
        if (state is State.Loaded) {
            if (state.podcast != null) {
                episodeManager.observeEpisodeByUuid(state.episode.uuid)
                    .combine<BaseEpisode, Podcast, State>(podcastManager.observePodcastByUuidFlow(state.podcast.uuid).distinctUntilChanged { t1, t2 -> t1.isUsingEffects == t2.isUsingEffects }) { episode, podcast ->
                        State.Loaded(episode, podcast, state.queue)
                    }
                    .catch { emit(State.Empty) }
            } else {
                episodeManager.observeEpisodeByUuid(state.episode.uuid)
                    .map<BaseEpisode, State> { State.Loaded(it, state.podcast, state.queue) }
                    .catch { emit(State.Empty) }
            }
        } else {
            flowOf(state)
        }
    }
}
```
and modified `MediaSessionManager#startObserving()` to include
```kotlin
playbackManager.upNextQueue.getChangesFlowWithLiveCurrentEpisode(episodeManager, podcastManager)
      .distinctUntilChanged { stateOne, stateTwo ->
          UpNextQueue.State.isEqualWithEpisodeCompare(stateOne, stateTwo) { episodeOne, episodeTwo ->
              episodeOne.uuid == episodeTwo.uuid && episodeOne.duration == episodeTwo.duration
          }
      }
      .combine(settings.useRssArtwork.flow) { state, useRssArtwork ->
          Pair(state, useRssArtwork)
      }
      .onEach {
          updateUpNext(it.first, it.second)
      }
      .catch { Timber.e(it) }
```
However, this causes the media notification to never appear. I implemented this same method and flow combining in `MainActivity#setupPlayerViews` in place of the `upNextQueueObservable` block but this causes the mini player to never be shown.

I did have to add new methods `observePodcastByUuidFlow` to `PodcastManager` and `observeByUuidFlow` to `PodcastDao`, but these methods are nearly identical to the existing `observePodcastByUuid` and `observeByUuid` methods respectively, but returning a `Flow` instead of a `Flowable`.

Reply:
I can't tell why a solution doesn't work without seeing it. The simplest thing to do is to convert final result of `upNextQueue` in `MediaSessionManager` with `asFlow()`. This way we can incrementally change each component to a `Flow` and not necessarily in this PR.

Reply:
Ah! I was missing a call to `.launchIn(this)`. All seems to be working as expected now, thanks.